lib/src/switch/Switch.tsx

```diff
@@ -27,25 +25,49 @@ const DxcSwitch = ({
   const [switchId] = useState(`switch-${uuidv4()}`);
   const labelId = `label-${switchId}`;
   const [innerChecked, setInnerChecked] = useState(defaultChecked ?? false);
+  const [hasLabel] = useState((label !== "" && label !== null && label !== undefined) ?? false);
+
   const colorsTheme = useTheme();
   const translatedLabels = useTranslatedLabels();
   const backgroundType = useContext(BackgroundColorContext);
+  const refTrack = useRef(null);
+
+  const handleOnKeyDown = (event: React.KeyboardEvent<HTMLDivElement>) => {
+    switch (event.keyCode) {
```

Comment:
Use Key instead of keyCode.
Also the space can be used to toggle the switch.

https://www.w3.org/WAI/ARIA/apg/example-index/switch/switch-button.html

Reply:
Done!